vault/data_source_ldap_dynamic_role_credentials.go

```diff
@@ -0,0 +1,167 @@
+				Type:        schema.TypeBool,
+				Computed:    true,
+				Description: "True if the duration of this lease can be extended through renewal.",
+			},
+			consts.FieldDistinguishedNames: {
+				Type:        schema.TypeList,
+				Computed:    true,
+				Description: "List of the distinguished names (DN) created.",
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+			consts.FieldPassword: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Password for the dynamic role.",
+				Sensitive:   true,
+			},
+			consts.FieldUsername: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Name of the dynamic role.",
+			},
+		},
+	}
+}
+
+func readLDAPDynamicCreds(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client, err := provider.GetClient(d, meta)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	mount := d.Get(consts.FieldMount).(string)
+	role := d.Get(consts.FieldRoleName).(string)
+	fullPath := fmt.Sprintf("%s/creds/%s", mount, role)
+
+	secret, err := client.Logical().ReadWithContext(ctx, fullPath)
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("error reading from Vault: %s", err))
+	}
+	log.Printf("[DEBUG] Read %q from Vault", fullPath)
+	if secret == nil {
+		return diag.FromErr(fmt.Errorf("no role found at %q", fullPath))
+	}
+
+	response, err := parseLDAPDynamicCredSecret(secret)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId(response.username)
+	if err := d.Set(consts.FieldLeaseID, response.leaseID); err != nil {
+		return diag.FromErr(err)
+	}
+	if err := d.Set(consts.FieldLeaseDuration, response.leaseDuration); err != nil {
+		return diag.FromErr(err)
+	}
+	if err := d.Set(consts.FieldLeaseRenewable, response.leaseRenewable); err != nil {
+		return diag.FromErr(err)
+	}
+	if err := d.Set(consts.FieldDistinguishedNames, response.distinguishedNames); err != nil {
+		return diag.FromErr(err)
+	}
+	if err := d.Set(consts.FieldPassword, response.password); err != nil {
+		return diag.FromErr(err)
+	}
+	if err := d.Set(consts.FieldUsername, response.username); err != nil {
+		return diag.FromErr(err)
+	}
+	return nil
+}
+
+type lDAPDynamicCredResponse struct {
+	leaseID            string
+	leaseDuration      string
+	leaseRenewable     bool
+	distinguishedNames []string
+	password           string
+	username           string
+}
+
+func parseLDAPDynamicCredSecret(secret *api.Secret) (lDAPDynamicCredResponse, error) {
+	var (
+		leaseID            string
+		leaseDuration      string
+		leaseRenewable     bool
+		distinguishedNames []string
+	)
+	if leaseIDRaw, ok := secret.Data[consts.FieldLeaseID]; ok {
+		leaseID = leaseIDRaw.(string)
+	}
+
+	if leaseDurationRaw, ok := secret.Data[consts.FieldLeaseDuration]; ok {
+		leaseDuration = leaseDurationRaw.(string)
+	}
+
+	if leaseRenewableRaw, ok := secret.Data[consts.FieldLeaseRenewable]; ok {
+		leaseRenewable = leaseRenewableRaw.(bool)
+	}
```

Comment:
These values are part of the `secret` structure and aren't nested under `secret.Data`. We can update this to be
```suggestion
	leaseID = secret.LeaseID
	leaseDuration = secret.LeaseDuration
	leaseRenewable = secret.Renewable
```